src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -803,7 +805,8 @@ primitive_inst::primitive_inst(network& network, program_node const& node, bool
     , _fused_mem_offset((_fused_mem_count > 0 && node.has_fused_dep()) ? node.get_first_fused_dep_idx() : 0)
     , _can_be_optimized(node.can_be_optimized())
     , _can_share_buffer(node.can_share_buffer())
-    , _is_constant(node.is_constant()) {
+    , _is_constant(node.is_constant())
+    , _is_output_event(is_any_user_cpu(node.get_users()) || node.is_output()) {
```

Comment:
I'm afraid this `_is_output_event` might mislead the meaning. What about more descriptive name, such as `needs_to_wait`? (not sure whether this is a proper name)

Reply:
Yeah, agree, what do you think about `_needs_event` or maybe even `_needs_completion_event`?

Reply:
looks good to me.